internal/controller/kyma/controller.go

```diff
@@ -274,7 +316,12 @@ func (r *Reconciler) replaceSpecFromRemote(ctx context.Context, controlPlaneKyma
 		}
 		return err
 	}
-	remote.ReplaceModules(controlPlaneKyma, remoteKyma)
+
+	if err := r.ValidateKymaSpec(remoteKyma); err != nil {
```

Comment:
If we do not validate kyma spec channel, but just let the reconciliation happens, anyway, if user given an invalid global channel, it will still have the error messages in each module status, that this module not exists in that invalid channel.

But have a validating logic here, it complicates the reconciliation, which is this huge update status logic you introduced [here](https://github.com/kyma-project/lifecycle-manager/pull/1748/files#diff-4d414901850a616d4e06a120615896904639aef683f0a4d8860abf65c76179d4R226-R237)

Reply:
I reverted the reconcile logic, but I am leaving this validation here. This has an advantage of providing more useful error message and it doesn't introduce any additional complication to the reconciliation logic.